src/serde.rs

```diff
@@ -1,35 +1,40 @@
 extern crate serde;
 extern crate std;
 
+use self::std::fmt;
 use self::std::prelude::v1::*;
 use self::serde::{de, Deserialize, Deserializer, Serialize, Serializer};
 
 use Uuid;
 
 impl Serialize for Uuid {
-    fn serialize<S: Serializer>(&self, serializer: &mut S) -> Result<(), S::Error> {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
         serializer.serialize_str(&self.hyphenated().to_string())
     }
 }
 
 impl Deserialize for Uuid {
-    fn deserialize<D: Deserializer>(deserializer: &mut D) -> Result<Self, D::Error> {
+    fn deserialize<D: Deserializer>(deserializer: D) -> Result<Self, D::Error> {
         struct UuidVisitor;
 
         impl de::Visitor for UuidVisitor {
             type Value = Uuid;
 
-            fn visit_str<E: de::Error>(&mut self, value: &str) -> Result<Uuid, E> {
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                write!(formatter, "a UUID")
```

Comment:
Change this to `"a UUID string"` to make the type information more helpful. For example in JSON if you try to read an integer as  UUID it will give you:

> invalid type: integer \`101\`, expected a UUID string at line x column y